### PR TITLE
Add source to sdk target dependency

### PR DESCRIPTION
**Pull request by rowwingman**
Resolves https://github.com/tuist/tuist/issues/837

### Short description 📝

This change adds source for sdk target's dependency. Xcode provide system sdk and developer sdk for unit tests, for example.

### Solution 📦

Add source parameter to SDK target dependency.

### CODE REVIEW

File: Sources/TuistCore/Models/Dependency.swift [hunk 1]
```diff
@@ -6,13 +6,27 @@ public enum SDKStatus {
     case optional
 }
 
+public enum SDKSource {
+    case developer
+    case system
+
+    public var path: String {
```
Comment (kwridan) [inline Sources/TuistCore/Models/Dependency.swift]: The logic that determines the path currently lives in `SDKNode.init`, perhaps we can create a helper function there to determine the path taking into account the platform as well (which may not be iOS).

File: Sources/ProjectDescription/TargetDependency.swift [hunk 2]
```diff
@@ -105,6 +111,8 @@ public enum TargetDependency: Codable, Equatable {
 
 extension SDKStatus: Codable {}
 
+extension SDKSource: Codable {}
```
Comment (pepibumur) [inline Sources/ProjectDescription/TargetDependency.swift]: I'd move the conformance to Coddle to the definition of the enum. I don't see the reason to keep this in a separate line. If we move it to the declaration of the enum we can see at a glance that SDKSource conforms the protocol. I'd also do the same with the `SDKStatus` enum.

File: Sources/ProjectDescription/TargetDependency.swift [hunk 3]
```diff
@@ -159,7 +168,8 @@ extension TargetDependency {
             self = .package(product: package)
         case "sdk":
             self = .sdk(name: try container.decode(String.self, forKey: .name),
-                        status: try container.decode(SDKStatus.self, forKey: .status))
```
Comment (pepibumur) [inline Sources/ProjectDescription/TargetDependency.swift]: I'd add a test to ensure that the encoding & decoding logic is right.

File: Sources/TuistCore/Models/Dependency.swift (hunk 1 above)
Comment (pepibumur) [reply Sources/TuistCore/Models/Dependency.swift]: Agreed

**Comment by pepibumur** [comment]
Hey @rowwingman, is it ok if I take this over?

**Comment by rowwingman** [comment]
> Hey @rowwingman, is it ok if I take this over?

Yeah, go ahead

**Comment by rowwingman** [comment]
@kwridan I simplified the implementation. @pepibumur I think, if you want you can pick up.
As an option, we can add `XCTest.framework` to unit tests and ui tests targets by default if it is missing

**Comment by rowwingman** [comment]
@pepibumur do you want to finish it?

**Comment by kwridan** [comment]
Thanks for the updates @rowwingman !

Testing the branch I am still getting `Cannot load underlying module for 'XCTest'` when using `.sdk(name: "XCTest.framework")` am I missing a step?

Will push a fixture to the branch to help us test this.

**Comment by pepibumur** [comment]
@rowwingman / @kwridan I rebased from master and finalized the PR. I ended up adding a new type of dependency, `.xctest` that developers can use from their manifests:

```swift
let target = Target(name: "MyTarget", dependencies: [.xctest])
```

Internally, it uses the `source` attribute that Kas suggested, but I decided not to expose it publicly because for de users it's not obvious whether a framework is a `system` or a `developer` framework. I'd rather keep that internally and if we see that there are more frameworks like `XCTest`, we can add some internal logic to determine the value of `source`. 

I ran the fixture locally and it works so hopefully, CI is happy about the changes too.

**Pull request merged**